From 3ded49b5afc24779070f244fbd5f6e9dd95b6721 Mon Sep 17 00:00:00 2001
From: Guy Nisan <gnisan@habana.ai>
Date: Thu, 22 Oct 2020 13:25:42 +0300
Subject: [PATCH] [SW-14676] fit: add missing start address

This is a fix for startaddr API which was left out on all platforms
besides Gaudi (which was part of the original patch).

Original patch:
https://gerrit.habana-labs.com/#/c/48837/

Change-Id: I820ff5c93f6bc74ac773de8b0f36e4e8e0ab74c6
Signed-off-by: Guy Nisan <gnisan@habana.ai>
---

diff --git a/goya/common.c b/goya/common.c
index af19f19..e8b2438 100644
--- a/goya/common.c
+++ b/goya/common.c
@@ -205,7 +205,7 @@
 		if (strcmp(image[i].arch, "arm64"))
 			continue;
 
-		image_entry = (void *)image[i].loadaddr;
+		image_entry = (void *)image[i].startaddr;
 		break;
 	}
 
diff --git a/goya2/common.c b/goya2/common.c
index 65c2300..4b4596a 100644
--- a/goya2/common.c
+++ b/goya2/common.c
@@ -414,7 +414,7 @@
 		if (strcmp(image[i].arch, "arm64"))
 			continue;
 
-		image_entry = (void *)image[i].loadaddr;
+		image_entry = (void *)image[i].startaddr;
 		break;
 	}
 
diff --git a/goya2_arc_gnu/common.c b/goya2_arc_gnu/common.c
index e2005db..760b938 100644
--- a/goya2_arc_gnu/common.c
+++ b/goya2_arc_gnu/common.c
@@ -145,8 +145,8 @@
 		if (strcmp(image[i].arch, "arm64"))
 			continue;
 
-		lsb = (u32)image[i].loadaddr;
-		msb = (u32)(image[i].loadaddr >> 32);
+		lsb = (u32)image[i].startaddr;
+		msb = (u32)(image[i].startaddr >> 32);
 		/* Set CPU0 Reset address to U-Boot */
 		regmap_writel(lsb, mmCPU_CA53_CFG_RST_ADDR_LSB_0);
 		regmap_writel(msb, mmCPU_CA53_CFG_RST_ADDR_MSB_0);
